src/VstsDemoBuilder/Controllers/EnvironmentController.cs

```diff
@@ -83,13 +83,15 @@ public string GetStatusMessage(string id)
         [SessonTimeout]
         public ContentResult GetTemplate(string TemplateName)
         {
-            string templatesPath = Server.MapPath("~") + @"\Templates\";
             string template = string.Empty;
-
-            if (System.IO.File.Exists(templatesPath + Path.GetFileName(TemplateName) + @"\ProjectTemplate.json"))
+            if (!TemplateName.Contains("."))
```

Comment:
What happens if the template name contains a "."? Is it invalid?

Reply:
I'm seeing this happen a few times - it doesn't look like a _direct_ fix to the "uncontrolled path" issue reported, but is it something else?

Reply:
I'm  validating for template name whether it contains (usually it doesn't) any period (one or more), if yes, this code block will simply return to calling script without moving to next line. This will be hit by the script (ajax request) and the parameter that will be sent from the script variable. This is basically a Template Name which we have in our ```TemplateSettings.json``` file.